### ISSUE TITLE
[Feature] Block accounts from signup

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#6356]
## Summary
Even if the user account is removed from an org, the user can still login and create their own orgs. We need a way to stop certain accounts from doing so. One way could be delete the accounts so that they can't login, but this wouldn't prohibit them from signing up again. Another way would be block certain accounts so that these accounts can never login / signup. 

```
That doesn't work.  I can manage members of my own organization,  but the newly self-registered users have created their own organizations,  and don't appear in mine.
```

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_66fv0)

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum the user can do this by setting APPSMITH_SIGNUP_DISABLED to true in their env

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal this issue was meant to target the use case where the admin wants to block out already signed up accounts. I think APPSMITH_SIGNUP_DISABLED env variable will cause all future signups to be disabled but won't affect the older signups.

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum , got it so this is to remove accounts from an instance?

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal yes, the use case here is to remove / block accounts from up and running instances.

**Comment by codedmind** [comment]
Maybe create a new role (App viewer) that only can use apps where they are invited... that way, if they can create orgs/apps shouldn't be needed to remove then...

**Comment by v4ld3rr4m4** [comment]
> @Nikhil-Nandagopal this issue was meant to target the use case where the admin wants to block out already signed up accounts. I think APPSMITH_SIGNUP_DISABLED env variable will cause all future signups to be disabled but won't affect the older signups.

**Comment by Nikhil-Nandagopal** [comment]
@samlander25 that's a good point. The basic user management at an instance level should actually be available in the community edition. We'll think about this internally and see how we can fit it in our roadmap.

**Comment by umbearto** [comment]
Totally agreeing with @samlander25.
Just imagine inviting someone from your company to your appsmith instance. After a while this person leaves the company...but there is no way to delete the account (only to un-share the different workspaces) and this same person could potentially still use the appsmith instance to create his own workspace\apps without the admin even being able to see the workspace.
I do understand having some feature behind a paywall, but this is basic user management.

**Comment by Nikhil-Nandagopal** [comment]
@umbearto fully agreed that this should not be behind a pay wall and we'll look into building it. We just haven't had a chance to get to it yet.

**Comment by acreacico** [comment]
For us this is also a concern. Would be great to have control over the users being able to access the platform unauthorized.

**Comment by abreland-amd** [comment]
We need the same functionality

**Comment by jack-goslin** [comment]
I need this as well. The form login setting is really confusing as it is a switch that changes the label, and I couldn't tell exactly what setting it was on. I accidentally had it set to allow anyone to sign up and now I'm not able to see if anyone did. Really worrying.

**Comment by GreenFlux** [comment]
Another user on Discord has requested this feature. [[thread]](https://discord.com/channels/725602949748752515/725609493974614076/1300951693571133470)
> Is basic user management/deactivation ever going to be added to the community version? All I want to do is be able to completely remove users and workspaces. It's really scary not being able to see users that have signed up or what they may have created, especially when they are employees that have been let go.

**Comment by theoroborus** [comment]
Certainly a feature I would like to see.

**Comment by Propelbi** [comment]
Very Important feature! We need to be able to delete users of people who no longer work with us. Please increase this feature's priority!

**Comment by Nikhil-Nandagopal** [comment]
Closing this as a duplicate of #21205
